api/components/text.schema.json

```diff
@@ -17,6 +17,32 @@
         },
         "style": {
             "$ref": "defs/textStyle.schema.json"
+        },
+        "locale": {
```

Comment:
In my opinion, it would be better if it was a schema in the defs folder. Also, you should define this as an object which has two properties : countryCode and languageCode (maybe scriptCode too) just as Flutter implements it.
https://api.flutter.dev/flutter/dart-ui/Locale-class.html